### PR TITLE
Add all edge-case unit tests and integration test for registry blacklist

### PR DESCRIPTION
**Pull request by amityadav0**
- Flag, Unflag tests
- i_am_human flagged tests
- Set authorised tests
- Add `assert_not_blacklisted` method
- Blacklisted user must not be able to perform soul transfer
- Blacklisted user must not be able to perform soul transfer[Add test]
- Migration test

### CODE REVIEW

File: contracts/registry/src/lib.rs [hunk 1]
```diff
@@ -2759,4 +2768,124 @@ mod tests {
             "{}".to_string(),
         );
     }
+
+    #[test]
+    fn admin_set_authorized_flaggers() {
+        let (mut ctx, mut ctr) = setup(&alice(), MINT_DEPOSIT);
```
Comment (sczembor) [inline contracts/registry/src/lib.rs:2774]:
```suggestion
        let (mut ctx, mut ctr) = setup(&admin(), MINT_DEPOSIT);
```

File: contracts/registry/src/lib.rs [hunk 2]
```diff
@@ -2759,4 +2768,124 @@ mod tests {
...
+
+        ctx.predecessor_account_id = admin();
+        testing_env!(ctx.clone());
```
Comment (sczembor) [inline contracts/registry/src/lib.rs:2777]:
```suggestion
```

File: contracts/registry/src/lib.rs [hunk 3]
```diff
@@ -2759,4 +2768,124 @@ mod tests {
...
+        let flaggers = [dan()].to_vec();
+        ctr.admin_set_authorized_flaggers(flaggers);
+       
+        ctx.predecessor_account_id = dan();
+        testing_env!(ctx.clone());
```
Comment (sczembor) [inline contracts/registry/src/lib.rs:2782]:
```suggestion
        testing_env!(ctx);
```

File: contracts/registry/src/lib.rs [hunk 4]
```diff
@@ -2759,4 +2768,124 @@ mod tests {
...
+
+        assert_eq!(ctr.account_flagged(dan()), Some(AccountFlag::Blacklisted));
+        assert_eq!(ctr.account_flagged(issuer1()), Some(AccountFlag::Blacklisted));
+        assert_eq!(ctr.account_flagged(issuer2()), Some(AccountFlag::Verified));
+    }
+
+    #[test]
+    #[should_panic(expected = "not authorized")]
+    fn admin_flag_accounts_non_authorized() {
+        let (mut ctx, mut ctr) = setup(&alice(), MINT_DEPOSIT);
```
Comment (sczembor) [inline contracts/registry/src/lib.rs:2802]: same as above, we can just pass the predecessor into to the setup and delete the two lines below

File: contracts/registry/tests/workspaces.rs [hunk 5]
```diff
@@ -236,6 +247,8 @@ async fn migration_mainnet() -> anyhow::Result<()> {
 }
 
 #[ignore = "this test is not valid after the migration"]
```
Comment (sczembor) [inline contracts/registry/tests/workspaces.rs:249]: I think this test cannot be ingnored since we need the migration after the last changes

**Review by sczembor — COMMENTED** [review]
left few suggestions. Lets update the migration test
